core/conversion/converters/impl/pooling.cpp

```diff
@@ -9,361 +10,222 @@ namespace converters {
+  bool expandDims = (orig_dims.nbDims < 4);
 
-  // Avg Pool needs at least 4D input
-  if (shape.size() < 4) {
-    auto new_shape = util::toDimsPad(shape, 4);
-    LOG_DEBUG(
-        "Input shape is less than 4D got: "
-        << util::toDims(shape) << ", inserting shuffle layer to reshape to 4D tensor shape: " << new_shape);
-    auto shuffle = ctx->net->addShuffle(*in);
-    shuffle->setReshapeDimensions(new_shape);
-    shuffle->setName((util::node_info(n) + " [Reshape to " + util::toStr(new_shape) + ']').c_str());
-    in = shuffle->getOutput(0);
+  if (expandDims) {
+    in = addPadding(ctx, n, in, 4, false, true);
   }
 
   auto kernel_size = util::toDims(args[1].unwrapToIntList());
-  LOG_DEBUG("kernel_size: " << kernel_size);
   auto padding = util::toDims(args[3].unwrapToIntList());
-  LOG_DEBUG("padding: " << padding);
   auto stride = util::toDims(args[2].unwrapToIntList());
-  if (args[2].unwrapToIntList().size() == 0) {
+  if (stride.nbDims == 0) {
     LOG_DEBUG("Stride not providied, using kernel_size as stride");
     stride = util::toDims(args[1].unwrapToIntList());
   }
+
+  if (kernel_size.nbDims == 1) {
+    kernel_size = util::unsqueezeDims(kernel_size, 0, 1);
+    LOG_DEBUG("kernel_size.nbDims < 2, padding:" << kernel_size);
+    LOG_DEBUG("kernel_size: " << kernel_size);
+  }
+  if (padding.nbDims == 1) {
+    padding = util::unsqueezeDims(padding, 0, 0);
+  }
+  if (stride.nbDims == 1) {
+    stride = util::unsqueezeDims(stride, 0, 1);
+  }
+  
+  LOG_DEBUG("kernel_size: " << kernel_size);
+  LOG_DEBUG("padding: " << padding);
   LOG_DEBUG("stride: " << stride);
 
-  bool ceil_mode = args[4].unwrapToBool();
-  bool count_inlcude_pad = args[5].unwrapToBool();
+  bool ceil_mode;
+  nvinfer1::IPoolingLayer* new_layer;
+
+  if (pool_type == nvinfer1::PoolingType::kMAX) {
+    auto dilation = util::toDims(args[4].unwrapToIntList());
+
+    TRTORCH_ASSERT(
+        dilation == util::toDims(std::vector<int64_t>(dilation.nbDims, 1)),
+        "Pooling dilation is not supported in TensorRT");
+
+    LOG_DEBUG("dilation: " << dilation);
+    LOG_WARNING("Dilation not used in Max pooling converter");
+    ceil_mode = args[5].unwrapToBool();
 
-  auto new_layer = ctx->net->addPoolingNd(*in, nvinfer1::PoolingType::kAVERAGE, kernel_size);
-  TRTORCH_CHECK(new_layer, "Unable to create Avg Pooling layer from node: " << *n);
+    new_layer = ctx->net->addPoolingNd(*in, pool_type, kernel_size);
+    TRTORCH_CHECK(new_layer, "Unable to create Max Pooling layer from node: " << *n);
+  } else if (pool_type == nvinfer1::PoolingType::kAVERAGE) {
+    ceil_mode = args[4].unwrapToBool();
+    bool count_inlcude_pad = args[5].unwrapToBool();
 
+    new_layer = ctx->net->addPoolingNd(*in, pool_type, kernel_size);
+    TRTORCH_CHECK(new_layer, "Unable to create Avg Pooling layer from node: " << *n);
+    new_layer->setAverageCountExcludesPadding(!count_inlcude_pad);
+  } else {
+    TRTORCH_ASSERT(0, "Unsupported pool mode!");
```

Comment:
We can use TRTORCH_THROW_ERROR for this instead of an always failing assert

Reply:
Fixed.